### ISSUE TITLE
Question + Issue

### ISSUE THREAD

**Issue by venaxyt** [Nuitka/Nuitka#1349]
Hi, I have a question about Nuitka, and an issue I don't know how to fix.
First the question: Can a Nuitka onefile executable be decompiled ?
Because for the moment I didn't find anything about people managing to do it.

Now the issue, I am trying to import a file in the onefile executable, that is needed to work correctly, it is a text file.
But when my program as executable compiled with Nuitka runs, I have the error:
Could not find "C:\Users\Temp.......text_file.txt".
So I first tought it was because of the file is in the executable I have to change in my main script open("text_file.txt", "r") by it's relative path, so I did that.
```py
import os, sys

def resource_path(relative_path):
    if hasattr(sys, "_MEIPASS"):
        return os.path.join(sys._MEIPASS, relative_path)
    return os.path.join(os.path.abspath("."), relative_path)

my_file = open(resource_path("text_file.txt"))
```
At this point it was still perfectly working in Python and I made that because it's the same with PyInstaller or Py2Exe.
But compiling it as an executable, I had the same error again.
Please if you could help me I would be very grateful, thank you !
(And also congratulations for Nuitka it's an incredible work)

**Comment by kayhayen** [comment]
Yes, it is trivial to put the onefile back to the pieces. It is essentially as secure as the standalone distribution folder, it is only a self extracting archive. With Nuitka Commercial https://nuitka.net/doc/commercial.html# you will that it can include data files into your program.

In the user manual your issue is already addressed: https://nuitka.net/doc/user-manual.html#onefile-finding-files using "." is not a good idea anyway, as programs can launch outside of their folders, and onefile does that. Instead you either use `sys.argv`  or `__file__` usually with `os.path.dirname`, both of which are a better idea for all code anyway.

**Comment by venaxyt** [comment]
Thank you very much for helping me, let me read that, please let the issue open for me to read and respond x)

**Comment by venaxyt** [comment]
Ok thanks I have understood, but am I stupid or I tought like in PyInstaller you can add file into the executable with --include-plugin-files="file.extension" ? And I am sorry to be so boring 😅 but please could you give me an example if it is possible, in python of a code which just opens and print in the console a text file in the executable itself ? I would be very grateful.
And maybe I'm gonig to buy the paid version but for the moment I really have not enough money for that x(

**Comment by venaxyt** [comment]
Ok so thanks for helping but this does not change, I replaced it by ``open(path.join(path.dirname(__file__), "text_file.txt"), "r")``
And i have the same error Could not find..... but I included it in the executable (with --include-plugin-files=text_file.txt) and it didn't show me any error in the console so all is good x(

**Comment by kayhayen** [comment]
It's not a plugin file, it's a data file to Nuitka, these have a set of options for how to include them.

**Comment by venaxyt** [comment]
Please could you send me an example command with data file to see ?

**Comment by venaxyt** [comment]
![image](https://user-images.githubusercontent.com/81310818/147964690-cf7ce23c-5521-4f43-89f1-dc7508e0a251.png)
...

**Comment by venaxyt** [comment]
Ok i found the command --include-data-file=path\\file.txt=file.txt
i have another error, i will say you ok

**Comment by venaxyt** [comment]
OH MY GOD FINALLY WORKING THANKS BRO
Another thing (btw +1 star xD), is it possible to import a folder ?

**Comment by venaxyt** [comment]
Ok sorry for making you waste your time i just found that yes let me test that

**Comment by venaxyt** [comment]
IT PERFECTLY WORKED THANK YOU !!!!

**Comment by venaxyt** [comment]
Thank you very much for your support !
